Data/ByteString/Internal.hs

```diff
@@ -648,6 +649,31 @@ concat = \bss0 -> goLen0 bss0 bss0
    concat [x] = x
  #-}
 
+-- | /O(log n)/ Repeats the given ByteString n times.
+times :: Integral a => a -> ByteString -> ByteString
+times n (BS fp len)
+  | n < 0 = error "stimes: non-negative multiplier expected"
+  | n == 0 = mempty
+  | len == 0 = mempty
```

Comment:
I wonder whether it might save allocations if we'd return the original bytestring in this case.

Reply:
On second thought, it's probably more important to free any pinned memory as soon as possible.